libPhoneNumber/NBGeocoderMetadataHelper.m

```diff
@@ -0,0 +1,132 @@
+              @"geocodingPairs%@ where NATIONALNUMBER in toSearch order by natLength desc limit 1",
+              countryCode] UTF8String],
+      -1, &self->selectStatement, NULL);
+  return self;
+}
+
+- (void)dealloc {
+  sqlite3_finalize(self->selectStatement);
+  sqlite3_close_v2(self->DB);
+}
+
+// returns a select statement for finding description given a phone number
+- (int)createSelectStatement:(NBPhoneNumber *)phoneNumber {
+  int sqliteResultCode;
+  @autoreleasepool {
+    sqliteResultCode = sqlite3_reset(self->selectStatement);
+    if (sqliteResultCode == SQLITE_OK) {
+      const char *completePhoneNumber =
+          [[NSString stringWithFormat:@"%@%@", phoneNumber.countryCode, phoneNumber.nationalNumber]
+              UTF8String];
+      sqlite3_bind_text(self->selectStatement, 1, completePhoneNumber, -1, SQLITE_TRANSIENT);
+      sqlite3_bind_text(self->selectStatement, 2, completePhoneNumber, -1, SQLITE_TRANSIENT);
+    } else {
+      NSLog(@"The prepare statement result code was: %d", sqliteResultCode);
+    }
+  }
+  return sqliteResultCode;
+}
+
+// returns YES if result found, NO if no results found
+- (NSString *)searchPhoneNumberInDatabase:(NBPhoneNumber *)phoneNumber
+                             withLanguage:(NSString *)language {
+  @autoreleasepool {
+    NSLog(@"In the searchPhoneNumberInDatabase class: %@, %@", phoneNumber, language);
+    if (![phoneNumber.countryCode isEqualToNumber:self.countryCode]) {
+      self.countryCode = phoneNumber.countryCode;
+      sqlite3_prepare_v2(
+          self->DB,
+          [[NSString
+              stringWithFormat:@"WITH RECURSIVE cnt(x) AS ( SELECT 1 UNION ALL SELECT x+1 FROM cnt "
+                               @"LIMIT length(?)), toSearch as (SELECT substr(?, 1, x) as indata "
+                               @"FROM cnt) select nationalnumber, description, "
+                               @"length(nationalnumber) as natLength from geocodingPairs%@ where "
+                               @"NATIONALNUMBER in toSearch order by natLength desc limit 1",
+                               [self.countryCode stringValue]] UTF8String],
+          -1, &self->selectStatement, NULL);
+      NSLog(@"CHANGED COUNTRY CODE FOR SEARCHING");
+    }
+
+    if (![self.language isEqualToString:language]) {
+      NSLog(@"CHANGED LANGUAGE CODE FOR SEARCHING");
+      sqlite3_close(self->DB);
+      self.language = language;
+      // grab bundle of current pod instance
+      NSBundle *bundle = [NSBundle bundleForClass:self.classForCoder];
+      NSURL *bundleURL = [[bundle resourceURL] URLByAppendingPathComponent:@"Resources.bundle"];
+      // create string for database directory
+      NSString *databasePath = [NSString stringWithFormat:@"%@%@.db", bundleURL, self.language];
+      self->databasePath = databasePath;
+      // open the database position
+      sqlite3_open([databasePath UTF8String], &self -> DB);
+    }
```

Comment:
See the comment on making this a readonly thing where language can't change in between.

Reply:
Yup, this has been updated as well.